qpc/insights/tests_insights_configure.py

```diff
@@ -0,0 +1,87 @@
+"""Test the CLI module."""
+
+import sys
+import unittest
+from io import StringIO
+
+from qpc.cli import CLI
+from qpc.tests_utilities import HushUpStderr, redirect_stdout
+from qpc.utils import read_insights_config, write_insights_config, write_server_config
+
+DEFAULT_PORT = 443
+DEFAULT_HOST = 'console.redhat.com'
+
+DEFAULT_CONFIG = {'host': '127.0.0.1', 'port': 8000, 'use_http': True, 'require_token': False}
+
+
+class InsightsConfigureTests(unittest.TestCase):
+    """Class for testing insights host configuration."""
+
+    def setUp(self):
+        """Create test setup."""
+        write_server_config(DEFAULT_CONFIG)
+        # Temporarily disable stderr for these tests, CLI errors clutter up
+        self.orig_stderr = sys.stderr
+        sys.stderr = HushUpStderr()
```

Comment:
Why is the "default" `write_server_config` is here?

Reply:
@nicolearagao this question remains open. Why setup is calling this?

Reply:
write_server_config is necessary due to a few validations qpc does during runtime:
`        if not is_server_config:
            # Before attempting to run command, check server location
            server_location = get_server_location()
            if server_location is None or server_location == '':
                log.error(_(messages.SERVER_CONFIG_REQUIRED % PKG_NAME))
                sys.exit(1)

        if read_require_auth():
            if ((not is_server_cmd or is_server_logout) and
                    not read_client_token()):
                log.error(_(messages.SERVER_LOGIN_REQUIRED % PKG_NAME))
                sys.exit(1)`